js/timepicker.js

```diff
@@ -485,6 +495,65 @@
       }
     }
 
+    _inputFromTextField() {
+      var isHours = this.currentView === 'hours';
+      var unit = Math.PI / (isHours ? 6 : 30);
+      var radius, radian;
+
+      if (this.options.twelveHour) {
+        radius = this.options.outerRadius;
+      }
+
+      if (isHours) {
+        var value = this['inputHours'].value;
+
+        if (!(value > 0 && value < 13)) {
+          this['inputHours'].value = new Date().getHours();
+        } else {
+          radian = value * unit;
+
+          var cx1 = Math.sin(radian) * (radius - this.options.tickRadius),
+            cy1 = -Math.cos(radian) * (radius - this.options.tickRadius),
+            cx2 = Math.sin(radian) * radius,
+            cy2 = -Math.cos(radian) * radius;
+
+          this.hand.setAttribute('x2', cx1);
+          this.hand.setAttribute('y2', cy1);
+          this.bg.setAttribute('cx', cx2);
+          this.bg.setAttribute('cy', cy2);
+
+          this.showView('minutes', this.options.duration / 2);
+
+          this.hours = value;
+
+          this.spanMinutes.focus();
+        }
+      } else {
+        var value = this['inputMinutes'].value;
+
+        if (!(value >= 0 && value < 60)) {
+          var minutes = new Date().getMinutes();
+          this['inputMinutes'].value = Timepicker._addLeadingZero(minutes);
+        } else {
```

Comment:
Same comments as previous ones : try to use const, invert the if to simplify the condition and if possible, refactor the code to move large blocks into separate methods.

Reply:
Agree same as above